include/nbl/asset/ICommandBuffer.h

```diff
@@ -0,0 +1,317 @@
+#ifndef __NBL_I_COMMAND_BUFFER_H_INCLUDED__
+#define __NBL_I_COMMAND_BUFFER_H_INCLUDED__
+
+#include <nbl/core/IReferenceCounted.h>
+#include "nbl/asset/IImage.h"
+#include "nbl/asset/IRenderpass.h"
+#include "nbl/asset/ISampler.h"
+#include "nbl/asset/ISpecializedShader.h"
+
+#include <type_traits>
+
+namespace nbl {
+namespace asset
+{
+
+//TODO move and possibly rename
+struct SBufferCopy
+{
+    size_t srcOffset;
+    size_t dstOffset;
+    size_t size;
+};
+struct SImageBlit
+{
+    asset::IImage::SSubresourceLayers srcSubresource;
+    asset::VkOffset3D srcOffsets[2];
+    asset::IImage::SSubresourceLayers dstSubresource;
+    asset::VkOffset3D dstOffsets[2];
+};
+struct SImageResolve
+{
+    asset::IImage::SSubresourceLayers srcSubresource;
+    asset::VkOffset3D srcOffset;
+    asset::IImage::SSubresourceLayers dstSubresource;
+    asset::VkOffset3D dstOffset;
+    asset::VkExtent3D extent;
+};
+struct SViewport
+{
+    float x, y;
+    float width, height;
+    float minDepth, maxDepth;
+};
+struct VkOffset2D
+{
+    int32_t x;
+    int32_t y;
+};
+struct VkExtent2D
+{
+    uint32_t width;
+    uint32_t height;
+};
+struct VkRect2D
+{
+    VkOffset2D    offset;
+    VkExtent2D    extent;
+};
```

Comment:
since we'll always include the vulkan headers (and dynamically link)... could we just use their structs?

Reply:
probably...
well, we always include volk but in backend code (not visible for `include/`)

Reply:
we could make Vulkan and OpenGL types always visible.

Reply:
i don't like this idea. Why? There's no way to make visible only structs without function declarations and enums/#defines

Reply:
I'm fine with making function declarations and enums visible.

Reply:
@achalpandeyy nuke the structs that duplicate Vulkan's (just download a KHR or Vulkan header with these structs in `3rdparty` then include it here)